src/screens/Home/controllers/HomeController.js

```diff
@@ -45,7 +39,11 @@ class HomeController {
     if (universityId) {
       const university = _.find(homeState.universities, { id: universityId });
       if (university) {
+        // reset selected departments and terms when university changes
+        homeState.setSelDepartments([]);
+        homeState.setSelTerms([]);
```

Comment:
Just curious, what caused the bug? Was this the key fix? Or are there other relevant changes in the refactored feed builder?

Reply:
Yes, this 2 lines fixed the bug. The problem was that the selected departments/terms are arrays of IDs, so when selecting a new university and its departments and terms are fetched, the old IDs in the selected departments/terms cannot match objects in the new fetched data, and this gonna cause an exception.

The feed builder is used to refactor the process of creating feed sections on the homepage, e.g. Starred Offerings and department sections...